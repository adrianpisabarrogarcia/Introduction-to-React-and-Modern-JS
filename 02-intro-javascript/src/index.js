//Desestructuración
//Asignación Desestructurante


const person = {
    name: 'Carlos',
    age: 30,
    key: 'Ironman'
}

//En react la mayoría de veces utilizamos los const
const {name:name2} = person;
const {name, age, key} = person;


console.log(person.name)
console.log(person.age)
console.log(person.key)


console.log(name2)
//is deprecated: use destructuring assignment instead
console.log(name)

