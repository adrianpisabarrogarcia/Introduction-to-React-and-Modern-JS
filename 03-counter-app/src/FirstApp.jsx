//rafc <- atajo para crear componentes

//import { Fragment } from "react"

const newMessage = 'Adrián' //<- esto es una variable que se usa en el componente
const newBoolean = true 
const newArray = [1,2,3,4,5,6,7,8,9]
const newObject = {
    message: "Hola Mundo",
    title: "Adrián P. G.",
}

const sayHello = (name) => `Hello ${name}!`


export const FirstApp = () => {
    //cuando tenga que volver a ejecutar para mostrar esta función,
    //vovlera a renderizar esta variable por eso no es recomedable usarla dentro de una función
    //sino que se debe usar en el componente (arriba ☝️)
    //const newMessage = 'Adrián' <- esto es una variable local que no se puede usar fuera de la función

    return(
        //<Fragment>
        <>
            <h2>First App</h2>
            <p>This is a paragraph 1</p>
            <p>This is a paragraph 2</p>
            <p>This is a paragraph 3</p>
            <p>This is a paragraph 4</p>
            <p>This is a paragraph 5</p>
            <p>This is a paragraph 6</p>
            <h1>{ 2 * 4 }</h1>
            <h1>{ newMessage }</h1>
            <h1>{ newBoolean }</h1>
            <h1>{ newArray }</h1>
            <code>{ JSON.stringify( newObject ) }</code>
            <h1>{ newObject.message }</h1>
            <h1>{ newObject.title }</h1>
            {/*
            comentario
            multilinea
            <h1>{ newObject.message }</h1>
            */}
            <h1 data-testid="test-title">{ sayHello('Adrián') }</h1>
        </>
        //</Fragment>
    )
}