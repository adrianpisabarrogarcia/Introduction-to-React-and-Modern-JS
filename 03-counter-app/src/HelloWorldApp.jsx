//option 1
/*
function App() {
    return (
        <div>
            <h1>Hello World!!!</h1>
        </div>
    )
}

export default App;
*/

// option 2
export const App = () => {
    return (
        <div>
            <h1>Hello World!!!</h1>
        </div>
    )
}

//option 3
//export const App = () => <h1>Hola Mundo!</h1>
