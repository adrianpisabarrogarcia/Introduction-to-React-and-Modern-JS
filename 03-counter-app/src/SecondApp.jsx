
export const SecondApp = ({ title = 'Hola Pepe!', subtitle, passNumber }) => {

    return (
        <>  
            <h3>{ title }</h3>
            <h3>{ subtitle }</h3>
            <h3>{ subtitle }</h3>
            <h3>{ passNumber }</h3>
        </>
    )
}
