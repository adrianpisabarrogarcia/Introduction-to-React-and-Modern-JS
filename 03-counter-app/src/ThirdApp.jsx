//añadir la dependencia en la terminal con yarn add prop-types, npm install prop-types
import PropTypes from "prop-types"

const ThirdApp = ({ title, subtitle }) => {
    
    //asi mejor no
    /*
    if (!title) {
        throw new Error('Title is required')
    }
    */

    return(
        <div>ThirdApp ➡️ { title} - { subtitle } </div>
    )
}

export default ThirdApp


ThirdApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}