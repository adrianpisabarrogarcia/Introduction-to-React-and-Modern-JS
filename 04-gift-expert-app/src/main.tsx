import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import { GiftExpertApp } from "./components/GiftExpertApp";

import "./styles.css";

createRoot(document.getElementById("root")!).render(
  <StrictMode>
    {/* titulo */}
    <GiftExpertApp />
    {/* Input */}

    {/* Listado de Gifs */}
  </StrictMode>
);
