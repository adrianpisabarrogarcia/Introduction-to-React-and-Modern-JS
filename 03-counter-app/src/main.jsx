import React from 'react'
import ReactDOM from 'react-dom/client'

//import App from './HelloWorldApp'
import { App } from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import { SecondApp } from './SecondApp'
import ThirdApp from './ThirdApp'
import { CounterApp } from './CounterApp'
import './styles.css';

// Eventos de React: https://es.reactjs.org/docs/events.html

ReactDOM.createRoot(document.getElementById('root'))
    .render(
        <React.StrictMode>
            <App />
            <FirstApp />
            <SecondApp subtitle='Hola, Amaia!!' passNumber={1 + 3} />
            <ThirdApp subtitle="Hola, Amaia!!" />
            <hr></hr>
            <button onClick={(event) => console.log(event)}>
                Ver objeto evento
            </button>
            <button onClick={(event) => console.log(event)}>
                +1
            </button>
            <hr></hr>
            <CounterApp value={2 + 2} />

        </React.StrictMode>
    )

