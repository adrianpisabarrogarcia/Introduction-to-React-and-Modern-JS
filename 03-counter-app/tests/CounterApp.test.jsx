import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";


describe('Pruebas en <CounterApp />', () => {

    const initialValue = 10;

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<CounterApp value={initialValue} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar el valor inicial sobre 100', () => {
        render(<CounterApp value={100} />)
        expect(screen.getByText(100)).toBeTruthy();
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100');
    })

    test('Debe incrementar con el botón +1', () => {
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();
    })

    test('Debe decrementar el con el botón -1', () => { 
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();
     })

    test('Debe de funcionar el botón de reset', () => { 
        render(<CounterApp value={initialValue} />)
        fireEvent.click(screen.getByRole('button', { name: 'btn-add' }));
        fireEvent.click(screen.getByRole('button', { name: 'btn-add' }));
        fireEvent.click(screen.getByRole('button', { name: 'btn-add' }));
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        expect(screen.getByText('10')).toBeTruthy();
     })
})