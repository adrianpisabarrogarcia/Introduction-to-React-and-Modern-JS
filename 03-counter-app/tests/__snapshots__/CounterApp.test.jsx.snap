// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    aria-label="btn-add"
  >
     +1 
  </button>
  <button
    aria-label="btn-substract"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
